fili-security/src/main/java/com/yahoo/bard/webservice/web/security/RoleDimensionApiFilterRequestMapper.java

```diff
@@ -0,0 +1,161 @@
+package com.yahoo.bard.webservice.web.security;
+
+import static com.yahoo.bard.webservice.web.security.SecurityErrorMessageFormat.DIMENSION_MISSING_MANDATORY_ROLE;
+
+import com.yahoo.bard.webservice.application.AbstractBinderFactory;
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+import com.yahoo.bard.webservice.data.config.ResourceDictionaries;
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionField;
+import com.yahoo.bard.webservice.util.StreamUtils;
+import com.yahoo.bard.webservice.web.ApiFilter;
+import com.yahoo.bard.webservice.web.ChainingRequestMapper;
+import com.yahoo.bard.webservice.web.DataApiRequest;
+import com.yahoo.bard.webservice.web.FilterOperation;
+import com.yahoo.bard.webservice.web.RequestValidationException;
+
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+import org.apache.commons.lang3.tuple.Triple;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+
+/**
+ * A request mapper that ensures that a user has at least one relevant role on a dimension and applies access filters
+ * based on roles for that user.
+ */
+public class RoleDimensionApiFilterRequestMapper extends ChainingRequestMapper<DataApiRequest> {
+
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractBinderFactory.class);
+
+    public static final String SECURITY_SIGNUP_MESSAGE_KEY = "security_signup_message";
+    public static final String DEFAULT_SECURITY_MESSAGE = "Your security settings do not permit access to this " +
+            "resource.  Please contact your user administration for access.";
+
+    private final static String UNAUTHORIZED_USER_MESSAGE = SYSTEM_CONFIG.getStringProperty(
+            SYSTEM_CONFIG.getPackageVariableName(SECURITY_SIGNUP_MESSAGE_KEY),
+            DEFAULT_SECURITY_MESSAGE
+    );
+
+    private final String unauthorizedHttpMessage;
+
+    Dimension dimension;
+    Map<String, Set<ApiFilter>> roleApiFilters;
+
+    /**
+     * Constructor.
+     *
+     * @param resourceDictionaries  The dictionaries to use for request mapping.
+     * @param dimension  The dimension whose roles are being matched
+     * @param roleApiFilters  ApiFilters by role for a given dimension
+     */
+    public RoleDimensionApiFilterRequestMapper(
+            ResourceDictionaries resourceDictionaries,
+            Dimension dimension,
+            Map<String, Set<ApiFilter>> roleApiFilters
```

Comment:
Here's a question: Rather than mapping from user role to a set of filters, do we instead want to map from user role to a function? I'm thinking something that takes the DataApiRequest, and `ContainerRequestContext`. Customers may want to do something more complex than a static collection of filters for each role.